### PR TITLE
Open sqlite-db in RW for reading

### PR DESCRIPTION
**Pull request by Stapelzeiger**
This fixes playback on macOS when the *.db3-shm, *.db3-wal don't exist yet. See #485

### CODE REVIEW

File: rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ SqliteWrapper::SqliteWrapper(
   if (io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_ONLY) {
     int rc = sqlite3_open_v2(
       uri.c_str(), &db_ptr,
-      SQLITE_OPEN_READONLY | SQLITE_OPEN_NOMUTEX, nullptr);
+      SQLITE_OPEN_READWRITE | SQLITE_OPEN_NOMUTEX, nullptr);
```
Comment (emersonknapp) [inline rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp:40]: Does this pose any danger of misuse? Would it be better to do a quick "open-close" to create the proper files, then still open as readonly?

Comment (Stapelzeiger) [reply rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp:40]: Good point.  I did a quick test, opening in RW, executing the first statement (`PRAGMA schema_version;`), closing it and reopening as RO works, and the *.db3-shm, *.db3-wal are created. I can implement it like that, but we're doubling the number of lines of code.

I'm wondering why it needs RW anyway. I'm not familiar with sqlite, so I don't understand how/why these WAL files are used. Maybe we are using it wrong?

**Comment by Karsten1987** [comment]
We might use it incorrectly. According to the [documentation](https://sqlite.org/wal.html#read_only_databases) older versions of sqlite3 did not support WAL in RO mode. The way I understand it, the proposed “best-practice” is to apply journal_mode DELETE before closing the connection after the last write. But we might be able to apply the same pragma when opening the db in RO.

**Comment by Stapelzeiger** [comment]
So you're suggesting to open the db in RW, set the journal_mode=DELETE, close and reopen as RO?

**Comment by Stapelzeiger** [comment]
@Karsten1987 something like the above? (92b51a4)

**Comment by gbiggs** [comment]
@Stapelzeiger What's the status of this PR? Is this still needed? If so we should move forward with getting it fixed and merged.

**Comment by MichaelOrlov** [comment]
Closing as will not done, since relevant issue has been closed. Please refer to the closing comment https://github.com/ros2/rosbag2/issues/485#issuecomment-1233269755

**Pull request closed without merging**